src/Illuminate/Support/helpers.php

```diff
@@ -635,32 +636,29 @@ function ends_with($haystack, $needles)
      */
     function env($key, $default = null)
     {
-        $value = getenv($key);
-
-        if ($value === false) {
-            return value($default);
-        }
-
-        switch (strtolower($value)) {
-            case 'true':
-            case '(true)':
-                return true;
-            case 'false':
-            case '(false)':
-                return false;
-            case 'empty':
-            case '(empty)':
-                return '';
-            case 'null':
-            case '(null)':
-                return;
-        }
-
-        if (($valueLength = strlen($value)) > 1 && $value[0] === '"' && $value[$valueLength - 1] === '"') {
-            return substr($value, 1, -1);
-        }
+        return (new ServerConstAdapter)
+            ->get($key)
+            ->map(function ($value) {
+                switch (strtolower($value)) {
+                    case 'true':
+                    case '(true)':
+                        return true;
+                    case 'false':
+                    case '(false)':
+                        return false;
+                    case 'empty':
```

Comment:
I am tempted to also remove handling for empty, since you can already do `FOO=""` or just `FOO=`, and phpdotenv will treat it as the empty string.

Reply:
I think that removing the "empty" will be an useless BC.